### ISSUE TITLE
everywhere take a trait parameter

### ISSUE THREAD

**Issue by fommil** [milessabin/shapeless#317]
imagine you have a sealed family and you want to apply an `everywhere` to a `trait`/coproduct rather than a `case class`/product. Right now, in 2.10.4 2.0.0 I'm getting silent failure :cry: 

Is this supposed to be supported? If so, I don't think it is... and if it isn't can you please add support in the 2.0.0 line because somebody has to use a non-standard build tool at work, which means that certain somebody can't add macro paradise for 2.1.

**Comment by milessabin** [comment]
In shapeless 2.1.0 and Scala 2.10.5 an option for you? If it is you'll be able to use the new -Yfundep-materialization flag instead of the macro paradise plugin.

A little bird tells me that that 2.10.5 will be out towards the end of this month ... snapshots are available now.

**Comment by fommil** [comment]
@milessabin yeah, I've looked into that. I even tried getting macro paradise to work with a non-standard build tool that you know and love :stuck_out_tongue: 

So is this now supported in 2.1?

I created another ticket because even case class stuff in 2.10 is completely broken for me. I've decided to just give up trying with a minimal test case if 2.1 supports it. I'm keen to compare my reflective visitor with `everywhere` and report back... both in terms of correctness and performance. (I have rather a lot of data)

I would almost recommend that the docs for 2.0 be updated to make it clear that `everywhere` is more like `someplaces`
